### PR TITLE
Convert base components to function components

### PR DESCRIPTION
**Pull request by Phidelux**
Hi,

I have converted most class components to function components. They seem to work as expected, however I will add some tests in the future. The remaining components and views are somehow tricky to convert.

Best regards,
Andi

### CODE REVIEW

**Comment by Phidelux** [comment]
This is not yet completed. I will ask you to review this as soon as I completed this task.

**Comment by AdrianMachado** [comment]
If you could upload a before and after screen capture video with the react dev tools re-render highlighting turned on, that'd be awesome. I'm hoping to see a savings in the number of renders we are doing - which will improve performance on lower end phones

**Comment by Phidelux** [comment]
I have converted all class components to function components. The only component not converted yet is the ContextProvider, for which I will have to dive deeper into the hooks API. For now I would like to ask you, Adrian, to review this. I will add a screen capture later.

File: src/components/ArrowListItem/ArrowListItem.jsx [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
-    const { isFocused } = this.state;
+    const [isFocused, setFocused] = useState(false);
 
     const itemCls = prefixCls;
     const iconCls = `${prefixCls}-icon-${isFocused ? 'focused' : 'unfocused'}`;
     const lineCls = `${prefixCls}-line`;
     const primaryCls = `${prefixCls}-primary`;
     const secondaryCls = `${prefixCls}-secondary ${secondary ? '' : 'hidden'}`;
 
+    const handleFocusChange = newFocused => {
```
Comment (AdrianMachado) [inline src/components/ArrowListItem/ArrowListItem.jsx]: `newFocused` -> `isNowFocused`

**Pull request merged**